src/librustc_hir/hir.rs

```diff
@@ -437,6 +437,29 @@ impl GenericParam<'hir> {
     }
 }
 
+pub trait NextTypeParamName {
+    fn next_type_param_name(&self) -> &'static str;
+}
+
+impl NextTypeParamName for &[GenericParam<'_>] {
+    fn next_type_param_name(&self) -> &'static str {
+        // This is the whitelist of possible parameter names that we might suggest.
+        let possible_names = ["T", "K", "L", "A", "B", "C", "D", "E", "F", "G", "H", "I", "J"];
```

Comment:
`K` and `L` are surprising, I would've expected `T`, `U`, `V` at the least, maybe some `X`, `Y`, `Z`.

Reply:
Does `["T", "U", "V", "X", "Y", "Z", "A", "B", "C", "D", "E", "F", "G"];` seem reasonable to you?

Reply:
Please address this in a different PR. This seems unrelated to this one.

Reply:
We already had that code in the codebase, I only moved it to a common place so that it could be used here too, otherwise we can suggest `fn foo<T, T>() {}` by accident.